src/Libraries/Microsoft.AspNetCore.Diagnostics.Middleware/README.md

```diff
@@ -1 +1,94 @@
-README
+# Microsoft.AspNetCore.Diagnostics.Middleware
+
+HTTP request diagnostics middleware for tracking latency and enriching and redacting log output.
+
+## Install the package
+
+From the command-line:
+
+```dotnetcli
+dotnet add package Microsoft.AspNetCore.Diagnostics.Middleware
+```
+
+Or directly in the C# project file:
+
+```xml
+<ItemGroup>
+  <PackageReference Include="Microsoft.AspNetCore.Diagnostics.Middleware" Version="[CURRENTVERSION]" />
+</ItemGroup>
+```
+
+## Usage Example
+
+### Tracking HTTP Request Latency
+
+These components enable tracking and reporting the latency of HTTP request processing.
+
+The services can be registered using the following methods:
+
+```csharp
+public static IServiceCollection AddRequestCheckpoint(this IServiceCollection services)
+public static IServiceCollection AddRequestLatencyTelemetry(this IServiceCollection services)
+public static IServiceCollection AddRequestLatencyTelemetry(this IServiceCollection services, Action<RequestLatencyTelemetryOptions> configure)
+public static IServiceCollection AddRequestLatencyTelemetry(this IServiceCollection services, IConfigurationSection section)
+```
+
+The middleware can be registered using the following methods:
+
+```csharp
+public static IApplicationBuilder UseRequestCheckpoint(this IApplicationBuilder builder)
+public static IApplicationBuilder UseRequestLatencyTelemetry(this IApplicationBuilder builder)
+```
+
+For example:
+
+```csharp
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddRequestLatencyTelemetry();
+builder.Services.AddRequestCheckpoint(options => { });
+
+var app = builder.Build();
+
+app.UseRequestCheckpoint();
+app.UseRequestLatencyTelemetry();
+```
+
+### HTTP Request Logs Enrichment and Redaction
+
+These components enable enriching and redacting ASP.NET Core's [HTTP request logs](https://learn.microsoft.com/aspnet/core/fundamentals/http-logging/).
+
+These APIs are only available for ASP.NET Core 8+.
```

Comment:
This is true for the whole package, right? Although by taking a look at the project looks like this is also supported in .NET 6. Does this comment mean that tracking request latency is supported in .NET 6+ but doing Enrichment and Redaction is only for 8.0?

Reply:
> Does this comment mean that tracking request latency is supported in .NET 6+ but doing Enrichment and Redaction is only for 8.0?

Correct.